### ISSUE TITLE
Setting XAxis type to 'value' or 'time' replaces X-Axis data

### ISSUE THREAD

**Issue by Erik-Handeland** [go-echarts/go-echarts#194]
I'm tried to create a line chart while also specify the data type as something other then "category". However whenever I to do the data I set via .SetXAxis() becomes replaced with the data in .AddSeries().

It displays correctly with  opts.XAxis{ Type: "category" }
<img src="https://user-images.githubusercontent.com/16601367/129433842-db7d580f-a6fb-4ef1-84a1-b8a88cf93ebf.png" width="400">
<img src="https://user-images.githubusercontent.com/16601367/129433847-a49aeb1c-c675-4e13-bc06-c6aff6121794.png" width="400">

But if I use opts.XAxis{ Type: "time" } or "value" it will display this:
<img src="https://user-images.githubusercontent.com/16601367/129433897-566bdccb-f953-4c04-a2b3-13c47f048e0f.png" width="400">
<img src="https://user-images.githubusercontent.com/16601367/129433899-acfeadce-f772-4baa-9bfd-2b6446ee6061.png" width="400">

Heres how I'm drawing my graph:
```
func lineChart(time []time.time, series []opts.LineData) *charts.Line {
	line := charts.NewLine()

	line.SetGlobalOptions(
		charts.WithTitleOpts(opts.Title{Title: "title"}),
		charts.WithXAxisOpts(opts.XAxis{
			Name: "Time",
			Type: "time",
		}),
		charts.WithYAxisOpts(opts.YAxis{
			Name: "Value",
			Type: "value",
		}),
		charts.WithDataZoomOpts(opts.DataZoom{Type: "slider"}),
		charts.WithTooltipOpts(opts.Tooltip{Show: true, Trigger: "axis"}),
		charts.WithLegendOpts(opts.Legend{Right: "80%"}),
	)

	line.SetXAxis(time).AddSeries("A", series)
	return line
}

```

**Comment by aranjan7** [comment]
+1 
I encountered this issue as well.

**Comment by free2k** [comment]
I also encountered this problem, and eventually I wanted to give up using it for this reason.

**Comment by free2k** [comment]
@Erik-Handeland Did you solve this problem？

**Comment by Erik-Handeland** [comment]
> @Erik-Handeland Did you solve this problem？

No, I ended up just keeping it as strings (category) which wasn't a deal breaker for my use case.

**Comment by aleksrosz** [comment]
@aranjan7 yes, thats what I mean. Compare HTML code when you use type category, value and timeseries. When you use timeseries and value there is bug and it is not working properly.

**Comment by aleksrosz** [comment]
@aranjan7  @Erik-Handeland  I fixed that for value type. Can you give me how you generate "series"? 


For scatter it should looke like that:

func generateScatterItems() []opts.ScatterData {
	items := make([]opts.ScatterData, 0)
	for i := 0; i < len(test); i++ {
		items = append(items, opts.ScatterData{
			Value:      []interface{}{data_forX[i].AV, data_forY[i].AVZ},
			Symbol:     "roundRect",
			SymbolSize: 5,
		})
	}
	return items
}

**Comment by LucaFilitz** [comment]
If there is anyone having a hard time with this:

What @aleksrosz is trying to explain is that when you use the time type it does not matter what you pass as an X Axis.
(you can do chart.SetXAxis(nil).AddSeries(...))

But because of that you have to pass a x y pair in the series values.
Instead of Value: Y use Value: []interface{}{X, Y}

Hope this helps.
